django/db/models/sql/query.py

```diff
@@ -1083,8 +1088,9 @@ def join_parent_model(self, opts, model, alias, seen):
 
     def add_annotation(self, annotation, alias, is_summary=False, select=True):
         """Add a single annotation expression to the Query."""
+        allow_join = self.model is not None
         annotation = annotation.resolve_expression(
-            self, allow_joins=True, reuse=None, summarize=is_summary
+            self, allow_joins=allow_join, reuse=None, summarize=is_summary
```

Comment:
This is untested.

Reply:
I am not sure how to test that one, as the only place I could find the allow_joins used required having a model to go there.

I also don't remember why it was modified. Should I just mock the resolve_expression and check `allow_joins` value?

Reply:
I tried setting it to True as it was before and no test failed. Maybe we should juste revert this part. @charettes , do you remember why it was modified?

Reply:
In all cases `resolve_expression()` will raise `FieldError` when no model is specified, before trying to join anything. IMO we can revert this change.